File: src/main/java/com/github/sheauoian/sleep/DbDriver.java
package com.github.sheauoian.sleep;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

import static com.github.sheauoian.sleep.Sleep.logger;

public class DbDriver {
    private static final DbDriver DRIVER = new DbDriver();
    private Connection connection;
    private Statement statement;
    public DbDriver() {
        try {
            Class.forName("org.sqlite.JDBC");
            this.connection = DriverManager.getConnection("jdbc:sqlite:database.db");
            this.statement = connection.createStatement();
        } catch (Exception e){
            logger.warning(e.getMessage());
        }
    }
    public Connection getConnection() {
        return connection;
    }
    public Statement getStatement() {
        return statement;
    }
    public static DbDriver singleton() {
        return DRIVER;
    }
    public void closeConnection(){
        try {
            if (this.connection != null) {
                this.connection.close();
            }
        } catch (SQLException ex) {
            System.out.println(ex.getMessage());
        }
    }
}

File: src/main/java/com/github/sheauoian/croissantsys/DbDriver.kt
package com.github.sheauoian.croissantsys

import org.sqlite.SQLiteConfig
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

open class DbDriver {
    fun close() {
        try {
            con.close()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    companion object {
        val instance: DbDriver = DbDriver()
        var con: Connection

        init {
            try {
                val config = SQLiteConfig()
                config.setDateStringFormat("yyyy-MM-dd HH:mm:ss")
                con = DriverManager.getConnection("jdbc:sqlite:database.db", config.toProperties())
            } catch (e: Exception) {
                throw RuntimeException(e)
            }
        }
    }
}